dist/squel.d.ts

```diff
@@ -0,0 +1,235 @@
+      setFields(fields: Object, options?: { ignorePeriodsForFieldNameQuotes?: boolean }): FlavourUpdate;
+      where(condition: string | Expression, ...args: any[]): FlavourUpdate;
+      order(field: string, direction?: boolean, ...args: any[]): FlavourUpdate;
+      limit(limit: number): FlavourUpdate;
+      offset(limit: number): FlavourUpdate;
+      returning(str: string): FlavourUpdate;
+      updateOptions(options: Object): FlavourUpdate;
+      registerValueHandler(type: any, handler: Handler): FlavourUpdate;
+      isNestable(): boolean;
+      clone(): FlavourUpdate;
+      toString(): string;
+      toParam(options?: { numberedParametersStartAt?: number }): { text: string, values: any[] };
+    }
+
+    interface GenericDelete<FlavourDelete extends SqlDelete> {
+      get(table: string): FlavourDelete;
+      from(table: string, alias?: alias): FlavourDelete;
+      join(name: string, alias?: alias, condition?: string): FlavourDelete;
+      left_join(name: string, alias?: alias, condition?: string): FlavourDelete;
+      right_join(name: string, alias?: alias, condition?: string): FlavourDelete;
+      outer_join(name: string, alias?: alias, condition?: string): FlavourDelete;
+      where(condition: string | Expression, ...args: any[]): FlavourDelete;
+      limit(limit: number): FlavourDelete;
+      order(field: string, direction?: boolean, ...args: any[]): FlavourDelete;
+      offset(limit: number): FlavourDelete;
+      updateOptions(options: Object): FlavourDelete;
+      registerValueHandler<T>(type: T|string, handler: Handler): FlavourDelete;
+      isNestable(): boolean;
+      clone(): FlavourDelete;
+      toString(): string;
+      toParam(options?: { numberedParametersStartAt?: number }): { text: string, values: any[] };
+    }
+
+    interface QueryBuilderOptions {
+      autoQuoteAliasNames?: boolean;
+      autoQuoteFieldNames?: boolean;
+      autoQuoteTableNames?: boolean;
+      customValueHandlers?: Handler[];
+      fieldAliasQuoteCharacter?: string;
+      nameQuoteCharacter?: string;
+      nestedBuilder?: boolean;
+      numberedParameters?: boolean;
+      numberedParametersStartAt?: number;
+      replaceSingleQuotes?: boolean;
+      separator?: string;
+      singleQuoteReplacement?: string;
+      stringFormatter?: (val: string) => string;
+      tableAliasQuoteCharacter?: string;
+    }
+
+    interface QueryBuilder {
+      select(options?: QueryBuilderOptions, blocks?: Object[]): SqlSelect;
+      insert(options?: QueryBuilderOptions, blocks?: Object[]): SqlInsert;
+      update(options?: QueryBuilderOptions, blocks?: Object[]): SqlUpdate;
+      delete(options?: QueryBuilderOptions, blocks?: Object[]): SqlDelete;
+      remove(options?: QueryBuilderOptions, blocks?: Object[]): SqlDelete;
+      expr(): Expression;
+      registerValueHandler<T>(type: T, handler: Handler): Squel;
+    }
+
+    // Flavour: None
+    interface SqlSelect extends GenericSelect<SqlSelect> {}
```

Comment:
Suggestion follow the names in the `.js` sources where possible, to aid searching (e.g. [`Select`](https://github.com/hiddentao/squel/blob/master/src/core.js#L2073), [`QueryBuilder`](https://github.com/hiddentao/squel/blob/master/src/core.js#L1946), [`BaseBuilder`](https://github.com/hiddentao/squel/blob/master/src/core.js#L208), etc)